devito/profiling.py

```diff
@@ -204,7 +204,9 @@ def summary(self, args, dtype, reduce_over=None):
             if grid is not None:
                 dimensions = (grid.time_dim,) + grid.dimensions
                 if all(d.max_name in args for d in dimensions):
-                    nt = args[grid.time_dim.max_name] - args[grid.time_dim.min_name] + 1
+                    max_t = args[grid.time_dim.max_name] or 0
```

Comment:
enlighten me please?

Reply:
well some can be None if there is a TimeFunction but no time loop like for `tests/test_operator.py test_staggered_time` where it just a single assign at a given index for a TImeFunction so this errors because `args[grid.time_dim.max_name]` is None

Reply:
this is exactly the bug I bumped into the other day, when I was accidentally running the test with DEVITO_LOGGING=DEBUG